servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -50,12 +52,18 @@
 
     @Nullable
     private ConnectionAcceptorFactory connectionAcceptorFactory;
-    @Nullable
-    private StreamingHttpServiceFilterFactory serviceFilter;
+    private final Deque<StreamingHttpServiceFilterFactory> noOffloadServiceFilters = new ArrayDeque<>();
+    private final Deque<StreamingHttpServiceFilterFactory> serviceFilters = new ArrayDeque<>();
```

Comment:
Originally I used `ArrayDeque` because there were additions at both ends which it handles better than `ArrayList`. If you don't need `List` interface and will be adding to both ends then `ArrayDeque` is more efficient.